src/PublicApiGenerator/EventNameBuilder.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.CodeDom;
+using System.Linq;
+using Mono.Cecil;
+
+namespace PublicApiGenerator
+{
+    public static class EventNameBuilder
+    {
+        public static string AugmentEventNameWithEventModifierMarkerTemplate(EventDefinition eventDefinition,
+            MemberAttributes addAccessorAttributes, MemberAttributes removeAccessorAttributes)
+        {
+            string name = eventDefinition.Name;
+            if (addAccessorAttributes != removeAccessorAttributes)
+            {
+                return name;
+            }
+
+            if (eventDefinition.DeclaringType.IsInterface)
+            {
+                return (addAccessorAttributes & MemberAttributes.VTableMask) == MemberAttributes.New
+                    ? string.Format(CodeNormalizer.EventModifierMarkerTemplate, $"new{CodeNormalizer.EventRemovePublicMarker}") + name
+                    : string.Format(CodeNormalizer.EventModifierMarkerTemplate, CodeNormalizer.EventRemovePublicMarker) + name;
+            }
+
+            bool? isNew = null;
+            var baseType = eventDefinition.DeclaringType.BaseType;
+            while (baseType != null)
+            {
+                var typeDef = baseType as TypeDefinition;
+                isNew = typeDef?.Methods.Any(e => e.Name.Equals(eventDefinition.AddMethod.Name, StringComparison.Ordinal));
+                if (isNew.HasValue && isNew.Value)
+                {
+                    break;
+                }
+                baseType = typeDef?.BaseType;
+            }
+
+            var addScopeAttributes = addAccessorAttributes & MemberAttributes.ScopeMask;
+            switch (addScopeAttributes)
+            {
+                case MemberAttributes.Static when addScopeAttributes == MemberAttributes.Static && !isNew.HasValue:
```

Comment:
Why one more `addScopeAttributes == MemberAttributes.Static` check here and below?